app.py

```diff
@@ -540,19 +556,35 @@ def gui_edit_member(member_id):
 
 @app.route("/gui/send_emails/")
 def gui_send_emails():
-    dirs = [obj for obj in os.listdir("emails")
-            if os.path.isdir("emails/{}".format(obj))
-            and obj != "__pycache__"
-            and obj != ".git"]
-    dirs.sort(reverse=True)
-    templates = { d: sorted(os.listdir("emails/{}".format(d))) for d in dirs }
-
-    # Strip the suffixes
-    for l in templates.values():
-        for i, t in enumerate(l):
-            l[i] = t[:t.find(".")]
-
-    return render_template("gui/send_emails.html", templates=templates)
+    # Get a list of all files in the "emails" directory.
+    files = subprocess.run(
+                "git ls-tree -r --name-only HEAD".split(),
+                cwd="emails",
+                capture_output=True
+            ).stdout.decode().split()
+
+    def last_changed(f):
+        """ Returns the unix timestamp of the last change to the file """
+        return int(subprocess.run(f"git log -1 --format=%at -- {f}".split(),
+                                  cwd="emails",
+                                  capture_output=True
+                   ).stdout.decode())
+
+    templates = [f for f in files if "/" in f and f.endswith(".tpl")]
```

Comment:
Maybe make this a filter.

Reply:
```suggestion
    templates = list(filter((lambda f: "/" in f and f.endswith(".tpl")), files))
```
Not sure if this is more readable. List comprehensions are quite powerful.